src/AuthorizedSocketConnection.js

```diff
@@ -81,19 +81,22 @@ export default class AuthorizedSocketConnection<TContext, TCredentials> {
     this.socket.emit('app_error', error);
   }
 
-  isAuthorized = (data: any) => {
```

Comment:
and then this can be the same as before, but:

```js
isAuthorized(
  data: any,
  hasPermission: (data: any, credentials: TCredentials) => boolean,
) {
  // ...
  const isAuthorized = !!credentials && hasPermission(data, credentials);
  // ...
}
```